File: solutionFiles/ContainerWithMostWater.js
const process = require('process');

function maxArea(height) {
    let left = 0;
    let right = height.length - 1;
    let maxArea = 0;

    while (left < right) {
        let width = right - left;
        let h = Math.min(height[left], height[right]);
        let currentArea = width * h;
        maxArea = Math.max(maxArea, currentArea);

        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }

    return maxArea;
}

if (process.argv.length !== 3) {
    console.error("Usage: node script.js '<json_string>'");
    process.exit(1);
}

try {
    const inputJson = process.argv[2];
    const data = JSON.parse(inputJson);
    const height = data.height;

    if (!Array.isArray(height)) {
        throw new Error("Invalid height data");
    }

    const result = maxArea(height);
    const outputJson = JSON.stringify({ result: result });

    console.log(outputJson);
} catch (error) {
    console.error("Invalid JSON input");
    process.exit(1);
}

File: solutionFiles/ContainerWithMostWater.java
import java.util.*;
import java.util.regex.*;

public class User1 {

    public static int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int maxArea = 0;

        while (left < right) {
            int width = right - left;
            int h = Math.min(height[left], height[right]);
            int currentArea = width * h;
            maxArea = Math.max(maxArea, currentArea);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxArea;
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java User1 '<json_string>'");
            System.exit(1);
        }

        String inputJson = args[0];
        try {
            // Extract the heights array from the JSON string
            Pattern pattern = Pattern.compile("\"height\":\\[(.*?)\\]");
            Matcher matcher = pattern.matcher(inputJson);
            if (!matcher.find()) {
                System.out.println("Invalid JSON input");
                System.exit(1);
            }

            String heightStr = matcher.group(1);
            String[] heightStrArray = heightStr.split(",");
            int[] height = new int[heightStrArray.length];
            for (int i = 0; i < heightStrArray.length; i++) {
                height[i] = Integer.parseInt(heightStrArray[i].trim());
            }

            // Compute the maximum area
            int result = maxArea(height);

            // Create the result JSON string
            String outputJson = String.format("{\"result\": %d}", result);
            System.out.println(outputJson);

        } catch (Exception e) {
            System.out.println("Invalid JSON input");
            System.exit(1);
        }
    }
}
